### DISCUSSION TITLE
Device linking signal states

### DISCUSSION THREAD

**Discussion by juliangiebel** [space-wizards/space-station-14#16994] [Technical]
This discussion is about #16992 
@deltanedas wrote:
> ## linking a non-momentary signal to something should immediately invoke it
>this would require port signal state to be stored in the port and not on whatever device's component
basically if i have a switch constantly emitting a high signal, if i link it to a door bolt i shouldnt have to toggle the switch twice to have it get high since its supposed to be a constant high signal
>
>same goes for the opposite, unlinking something that was high should invoke with lo

**Comment by juliangiebel** [comment]
Should the state be saved on the source port then? And the device link system just triggers the sink with the saved state when connecting?

**Reply by deltanedas** [reply]
state for linking should be stored on the source port yes

not sure which for unlinking

**Reply by juliangiebel** [reply]
Unlinking should either just send the opposite state or always low 🤔 
I think always low would make more sense

**Reply by deltanedas** [reply]
yeah always low is good
